resdk/resources/permissions.py

```diff
@@ -269,3 +276,43 @@ def __repr__(self):
                 ))
 
         return '\n'.join(res)
+
+    def _get_holders_with_perm(self, perm):
+        """Get Users/Group/public names that have ``perm`` perm."""
+        # Prevent circular imports
+        from .user import Group, User
+
+        self.fetch()
+
+        holders = []
+        for item in self._permissions:  # pylint: disable=protected-access
+            if perm in item['permissions']:
+                if item['type'] == 'user':
+                    holders.append(
+                        User(
+                            self.resolwe,
+                            id=item['id'],
+                            first_name=item['name'],
+                            username=item['username'],
+                        ),
+                    )
+                elif item['type'] == 'group':
+                    holders.append(Group(self.resolwe, id=item['id'], name=item['name']))
+                else:
+                    holders.append(User(self.resolwe, username='public', first_name="Public"))
+        return holders
+
+    @cachedproperty
```

Comment:
Why not? This is not a random third party library. It is 10 times more popular than for example `slumber` package that we heavily use. And why do you think it complicates the code? I see it as a very nice simplification. The aim here was to try it and see if it works OK. Then we can apply this pattern to other parts of the code.

Reply:
Each such addition to the codebase increase barrier to entry and multiplies the chance that something gets wrong. It is definitely useful to check such implementations and reuse bits of them, but even this must be done with care as it can get complicated very quickly (e.g. modifying the `__dict__`).

Reply:
Thanks for this feedback. You've convinced me :-) I've rewritten the code to mimic the current style of caching.